pkg/util/k8s/k8s.go

```diff
@@ -102,8 +102,15 @@ func (r ResourceManager) Secret(name, namespace string, secType corev1.SecretTyp
 }
 
 // Service returns a Service object.
-func (r ResourceManager) Service(name, namespace string, annotations map[string]string, spec *corev1.ServiceSpec) *resource.Service {
-	return resource.NewService(r.client, name, namespace, r.labels, annotations, spec)
+func (r ResourceManager) Service(name, namespace string, labels map[string]string, annotations map[string]string, spec *corev1.ServiceSpec) *resource.Service {
+	// Combine the common labels and resource specific labels.
+	if labels == nil {
+		labels = map[string]string{}
+	}
+	for k, v := range r.labels {
+		labels[k] = v
+	}
```

Comment:
Introduced label passing for Service only here. Will add it to other resources as well in separate PR in order to support recommended labels for various resources https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/ .